core/trino-main/src/main/java/io/trino/execution/scheduler/StageTaskSourceFactory.java

```diff
@@ -905,7 +905,8 @@ private static ListMultimap<PlanNodeId, ExchangeSourceHandle> getInputsForRemote
         ImmutableListMultimap.Builder<PlanNodeId, ExchangeSourceHandle> result = ImmutableListMultimap.builder();
         for (RemoteSourceNode remoteSource : remoteSources) {
             for (PlanFragmentId fragmentId : remoteSource.getSourceFragmentIds()) {
-                Collection<ExchangeSourceHandle> handles = requireNonNull(exchangeSourceHandles.get(fragmentId), () -> "exchange source handle is missing for fragment: " + fragmentId);
+                Collection<ExchangeSourceHandle> handles = exchangeSourceHandles.get(fragmentId);
+                checkArgument(!handles.isEmpty(), "exchange source handle is missing for fragment: %s", fragmentId);
```

Comment:
I'd actually drop that. Previous version wouldn't throw on empty collection - this does. It's a change in the behavior.

I've asked @losipiuk and according to his understanding dropping an RNN is a correct one. We could ask @arhimondr for an opinion though

Reply:
Sure, I can drop that. It just looked like it's the intent of the original code.

Reply:
I do not see a reason why we should expect non-empty handles here. I may be wrong though. @arhimondr ?

Reply:
I would probably just drop this check

Reply:
dropped